### PR TITLE
Add SYSTEM option

### LINKED ISSUE #377: heads up: CMake 3.25 `SYSTEM` option for FetchContent
[Here's the motivating issue in CMake's issue tracker](https://gitlab.kitware.com/cmake/cmake/-/issues/22401). `FetchContent` uses `ExternalProject_Add`, which uses `add_subdirectory`, which is unlike importing via `find_package` in that the targets that get added will not be `IMPORTED`, and will therefore have their `target_include_directories` not marked as `SYSTEM`, which means compilers will check those third party headers to warn for things that the user only wanted warnings for for their own project's code (not for third party header code).

[Here's the ticket tracking the solution that was proposed and is now being implemented](https://gitlab.kitware.com/cmake/cmake/-/issues/18040#note_1197146). In summary,

> 1. Add a SYSTEM property for targets
> 2. ...
> 3. As a helper based on the `SYSTEM` property, implement `add_subdirectory(.. SYSTEM)` and `FetchContent_Declare(.. SYSTEM BOOL)`

At the time of this writing, these changes are not merged in yet, and I'm not sure, but I don't think CMake 3.25 will be released until November, so it's still quite early to be mentioning this. I thought it would be nice to just put it on the radar so it doesn't go undetected.

I'm guessing there will be some policy variable that can be checked to see if the CPM user's CMake version supports these new arguments, and if so, to add a similar parameter/option to `CPMAddPackage`.

### PR DESCRIPTION
**Pull request by TheLartians**
Adds experimental support for the [SYSTEM](https://cmake.org/cmake/help/latest/prop_dir/SYSTEM.html#prop_dir:SYSTEM) property as an opt-in option for `CPMAddPackage`.

## Todo

- [x] Document the option in the Readme

## Open Questions

- Should it be on by default for the single argument syntax?
- As we now control the CMake version in the tests, should we also test legacy versions for backwards compatibility?

Something noteworthy from the [CMake docs](https://cmake.org/cmake/help/latest/command/target_include_directories.html) is that it changes the include search order.

> Additionally, system include directories are searched after normal include directories regardless of the order specified.

Closes #377

### CODE REVIEW

**Comment by TheLartians** [comment]
I've added a simple test case that should work if the property works as I expect,  but it doesn't seem to have the desired behaviour.

**Comment by ClausKlein** [comment]
> I've added a simple test case that should work if the property works as I expect, but it doesn't seem to have the desired behaviour.

Which CMake version are you using, its need **v3.25** and you have to include the header with `#include <>`

**Comment by iboB** [comment]
> Should it be on by default for the single argument syntax?

I think so. I'd even suggest to have this a default in the regular syntax.

> As we now control the CMake version in the tests, should we also test legacy versions for backwards compatibility?

Again I would recommend this as a yes. At least while Ubuntu 20.04 is still widely used. This is a potentially big user base with a CMake 3.16 as a default.

On a side note, I would suggest the following minimal supported version for CPM: the default CMake version of the second-latest Ubuntu LTS release currently available (the latest now is 22.04 and the second is 20.04 with CMake 3.16)

**Review by iboB — APPROVED** [review]
Other than the comment above, this looks pretty good!

**Comment by TheLartians** [comment]
> I think so. I'd even suggest to have this a default in the regular syntax.

I would actually prefer keeping it explicit in the explicit syntax. For one it keeps it consistent with `EXCLUDE_FROM_ALL` and additionally it allows using CPM as an included guarded `add_subdirectory` as I do in the [ModernCppStarter](https://github.com/TheLartians/ModernCppStarter/blob/84ed75ca7db28672a29d865382389ad3e0f03f8a/standalone/CMakeLists.txt#L19). 

> Again I would recommend this as a yes. At least while Ubuntu 20.04 is still widely used. This is a potentially big user base with a CMake 3.16 as a default. 
>
> On a side note, I would suggest the following minimal supported version for CPM: the default CMake version of the second-latest Ubuntu LTS release currently available (the latest now is 22.04 and the second is 20.04 with CMake 3.16)

Agree with this approach. I will make an update to the actions for legacy support in a separate PR!

File: cmake/CPM.cmake [hunk 1]
```diff
@@ -950,6 +950,10 @@ function(
     else()
       set(addSubdirectoryExtraArgs "")
     endif()
+    if(${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.25")
+      # https://cmake.org/cmake/help/latest/prop_dir/SYSTEM.html#prop_dir:SYSTEM
+      set(addSubdirectoryExtraArgs ${addSubdirectoryExtraArgs} SYSTEM)
```
[the remark this answers is not in the document]
Comment (ClausKlein) [reply cmake/CPM.cmake]: sorry, this review was a misstake!

**Pull request merged**